code/modules/urist/modules/shipbattles/torpedo_warhead.dm

```diff
@@ -0,0 +1,194 @@
+
+	else if(istype(I, /obj/item/weapon/wirecutters))
+		if(is_rigged && attached_device)
+			to_chat(user, "<span class='notice'>You carefully begin to remove [attached_device] from the warhead's internals.</span>")
+			if(do_after(user,4 SECONDS))
+				to_chat(user, "<span class='notice'>You carefully remove [attached_device] from the warhead's internals.</span>")
+				var/obj/item/device/assembly/A = attached_device
+				A.forceMove(get_turf(user))
+				A.holder = null
+				attached_device = null
+				is_rigged = 0
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+				icon_state = "torpedowarhead-open-mod"
+		else if(riggedstate == CIRCUITRY_RIGGED)
+			to_chat(user, "<span class='notice'>You snip the wire attached to the warhead's detonation circuit.</span>")
+			riggedstate = CIRCUITRY_MODIFIED
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open-mod"
+		else if(riggedstate == CIRCUITRY_MODIFIED)
+			to_chat(user, "<span class='notice'>You carefully undo the modifications to the warhead's circuitry.</span>")
+			riggedstate = CIRCUITRY_EXPOSED
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open"
+		else if(riggedstate == CIRCUITRY_EXPOSED)
+			to_chat(user, "<span class='notice'>You begin to carefully modify the circuitry of the warhead.</span>")
+			if(do_after(user,4 SECONDS))
+				to_chat(user, "<span class='notice'>You have modified the torpedo warhead's internal circuitry. It can now be wired up and attached to something.</span>")
+				riggedstate = CIRCUITRY_MODIFIED
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+				icon_state = "torpedowarhead-open-mod"
+		return
+	else if(istype(I, /obj/item/device/multitool))
+		if(riggedstate < CIRCUITRY_EXPOSED)
+			wires.Interact(user)
+	else if(istype(I, /obj/item/stack/cable_coil) && riggedstate == CIRCUITRY_MODIFIED)
+		to_chat(user, "<span class='notice'>You rig a wire from the torpedo warhead's detonator circuit. You can now attach something to it to detonate it remotely.</span>")
+		riggedstate = CIRCUITRY_RIGGED
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+	else if(istype(I, /obj/item/device/assembly))
+		if(riggedstate == CIRCUITRY_RIGGED)
+			var/obj/item/device/assembly/A = I
+			to_chat(user, "<span class='warning'>You rig the [A] to the torpedo warhead's detonator circuit!</span>")
+			is_rigged = 1
+			if(!user.unEquip(A, src))
+				return
+			A.forceMove(src)
+			attached_device = I
+			A.holder = src
+			log_and_message_admins("[user] has rigged a torpedo IED.")
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open-mod-armed"
+	return
+
+/obj/item/shipweapons/torpedo_warhead/attack_self(mob/user as mob)
+	..()
+	if(riggedstate < CIRCUITRY_EXPOSED)
+		wires.Interact(user)
+
+/obj/item/shipweapons/torpedo_warhead/proc/process_activation() // uh oh, time to boom
```

Comment:
I spy an unused function.

Reply:
Process activation is used for signallers.